src/Lykke.Service.Assets.Services/AssetGroupService.cs

```diff
@@ -151,18 +144,22 @@ public async Task RemoveClientFromGroupAsync(string clientId, string groupName)
 
         public async Task RemoveGroupAsync(string groupName)
         {
-            foreach (var assetId in await GetAssetIdsForGroupAsync(groupName))
-            {
-                await _assetGroupAssetLinkRepository.RemoveAsync(assetId, groupName);
-            }
+            var tasks = new List<Task>();
+            var assetLinksTasks = (await GetAssetIdsForGroupAsync(groupName)).Select(assetId =>
+                _assetGroupAssetLinkRepository.RemoveAsync(assetId, groupName));
+            tasks.AddRange(assetLinksTasks);
 
-            foreach (var clientId in await GetClientIdsForGroupAsync(groupName))
-            {
-                await _clientAssetGroupLinkRepository.RemoveAsync(clientId, groupName);
-                await _assetGroupClientLinkRepository.RemoveAsync(clientId, groupName);
-            }
+            var clientIds = (await GetClientIdsForGroupAsync(groupName)).ToArray();
+            var clientAssetGroupTasks =
+                clientIds.Select(clientId => _clientAssetGroupLinkRepository.RemoveAsync(clientId, groupName));
+            tasks.AddRange(clientAssetGroupTasks);
+            var assetGroupTasks =
+                clientIds.Select(clientId => _assetGroupClientLinkRepository.RemoveAsync(clientId, groupName));
+            tasks.AddRange(assetGroupTasks);
+
+            tasks.Add(_assetGroupRepository.RemoveAsync(groupName));
+            await Task.WhenAll(tasks.ToArray());
```

Comment:
Please, see the comment [below](https://github.com/LykkeCity/Lykke.Service.Assets/pull/53/files#r217013674).

Reply:
The order doesn't matter here. The only problem we can have here - the lack of transactionality. But I don't want to fix it right now. My goal is to reduce requests latency.